test/integration/controller/jobset_controller_test.go

```diff
@@ -96,11 +101,23 @@ var _ = ginkgo.Describe("JobSet validation", func() {
 
 					// If we are expecting a validation error, we can skip the rest of the checks.
 					if !update.shouldSucceed {
-						gomega.Expect(k8sClient.Update(ctx, js)).To(gomega.Not(gomega.Succeed()))
+						gomega.Eventually(func() error {
+							var jsGet jobset.JobSet
+							if err := k8sClient.Get(ctx, types.NamespacedName{Name: js.Name, Namespace: js.Namespace}, &jsGet); err != nil {
+								return err
+							}
+							return k8sClient.Update(ctx, &jsGet)
+						}, timeout, interval).ShouldNot(gomega.Succeed())
```

Comment:
hmm, we expect this update to not succeed , so lets leave this alone.

Reply:
actually, there is only a single test case that doesn't set this to false, so we can remove this code and `shouldSucceed` as well

Reply:
Removed this function and the flag.